server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/DatafileTransaction.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.tserver.tablet;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import org.apache.accumulo.core.metadata.StoredTabletFile;
+
+public abstract class DatafileTransaction {
+
+  protected final long ts = System.currentTimeMillis();
+
+  public void apply(Set<StoredTabletFile> files) {}
+
+  public Date getDate() {
+    return Date.from(Instant.ofEpochMilli(ts));
```

Comment:
This makes a new object per invocation and since this is used by hashCode(), adding a DatafileTransaction to a HashMap or HashSet will generate an object that will be immediately GC'd. Probably not a huge deal but we could  cache the date object or swap the `hashCode()` method to use `Long.hashCode(ts)`.

Reply:
done.